### PR TITLE
deprecate go-libp2p-core and use go-libp2p instead

### PR DESCRIPTION
**Pull request by nisainan**
glad to see libp2p is uniting repositories together and be happy to join it.

- deprecate go-libp2p-core and use go-libp2p instead
- replace some deprecated funtions with new
- annotation formated

### CODE REVIEW

File: peerdiversity/filter.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (f *Filter) TryAdd(p peer.ID) bool {
 	// don't allow peers for which we can't determine addresses.
 	addrs := f.pgm.PeerAddresses(p)
 	if len(addrs) == 0 {
-		dfLog.Debugw("no addresses found for peer", "appKey", f.logKey, "peer", p.Pretty())
+		dfLog.Debugw("no addresses found for peer", "appKey", f.logKey, "peer", p.String())
```
Comment (Stebalien) [inline peerdiversity/filter.go:156]: No need to call `String` explicitly.

File: generate/main.go [hunk 2]
```diff
@@ -66,5 +66,5 @@ func main() {
 		printf("%d,", j)
 	}
 	printf("\n}")
-	f.Close()
+	_ = f.Close()
```
Comment (Stebalien) [inline generate/main.go:69]: We should panic if this errors.

File: util.go [hunk 3]
```diff
@@ -2,15 +2,15 @@ package kbucket
...
-	"github.com/libp2p/go-libp2p-core/peer"
 	ks "github.com/libp2p/go-libp2p-kbucket/keyspace"
+	"github.com/libp2p/go-libp2p/core/peer"
 
 	u "github.com/ipfs/go-ipfs-util"
-	"github.com/minio/sha256-simd"
 )
 
-// Returned if a routing table query returns no results. This is NOT expected
+// ErrLookupFailure Returns if a routing table query returns no results. This is NOT expected
```
Comment (Stebalien) [inline util.go:13]:
```suggestion
// ErrLookupFailure is returned if a routing table query returns no results. This is NOT expected
```

File: peerdiversity/filter.go (hunk 1 above)
Comment (Stebalien) [reply peerdiversity/filter.go:156]: (saves some time if debug logging isn't enabled)

**Review by Stebalien — CHANGES_REQUESTED** [review]
Some small changes, otherwise LGTM.

And thanks! Making mistakes is all a part of the master plan so we can fix them and make everyone happy 😛.

File: peerdiversity/filter.go (hunk 1 above)
Comment (nisainan) [reply peerdiversity/filter.go]: removed 'peer'

File: generate/main.go (hunk 2 above)
Comment (nisainan) [reply generate/main.go]: panic added

File: util.go (hunk 3 above)
Comment (nisainan) [reply util.go]: resolved

**Comment by nisainan** [comment]
thanks for your advice and tolerance, hoping I can do better next time.

File: peerdiversity/filter.go [hunk 4]
```diff
@@ -153,7 +153,7 @@ func (f *Filter) TryAdd(p peer.ID) bool {
 	// don't allow peers for which we can't determine addresses.
 	addrs := f.pgm.PeerAddresses(p)
 	if len(addrs) == 0 {
-		dfLog.Debugw("no addresses found for peer", "appKey", f.logKey, "peer", p.Pretty())
+		dfLog.Debugw("no addresses found for peer", "appKey", f.logKey)
```
Comment (Stebalien) [inline peerdiversity/filter.go:156]: Ah, no, I meant:

```suggestion
		dfLog.Debugw("no addresses found for peer", "appKey", f.logKey, "peer", p)
```

**Comment by Stebalien** [comment]
> thanks for your advice and tolerance, hoping I can do better next time.

Nah, you're code was great.

File: peerdiversity/filter.go (hunk 4 above)
Comment (nisainan) [reply peerdiversity/filter.go]: got it

**Review by Stebalien — APPROVED** [review]

**Pull request merged**